pkg/config/matcher.go

```diff
@@ -28,3 +101,88 @@ func matchByAttributes(actual *processAttrs, required *Attributes) bool {
 	}
 	return true
 }
+
+func restartObjectWithAttributes(ctx context.Context, clientSet kubernetes.Interface, required *Attributes) error {
+	// pod labels is not compatible with metadata - unless it's namespace
+	// is namespace always required? does it need to be a simple string?
+	labels := required.PodLabels
+	namespace := "default"
+	ns := required.Metadata[AttrNamespace]
+	if ns != nil {
+		namespace = ns.re.String()
+	}
+
+	for attrName := range required.Metadata {
+		switch attrName {
+		case AttrNamespace:
+		case AttrPodName:
+			continue
+		}
+		if len(labels) > 0 {
+			return errors.New("pod labels is not compatible with metadata")
+		}
+	}
+
+	for attrName, criteriaRegexp := range required.Metadata {
+		switch attrName {
+		case AttrNamespace:
+		case AttrPodName:
+		case AttrDeploymentName:
+			err := restartDeployment(ctx, clientSet, namespace, criteriaRegexp)
+			if err != nil {
+				return err
+			}
+		case AttrReplicaSetName:
```

Comment:
If these are the same, should we just list them in the case above?

Reply:
we need a different k8s api call to query - so it's "analogous", not "same"